nvflare/app_common/workflows/hierarchical_statistics_controller.py

```diff
@@ -0,0 +1,335 @@
+            writer_id,
+            wait_time_after_min_received,
+            result_wait_timeout,
+            precision,
+            min_clients,
+            enable_pre_run_task,
+        )
+        self.hierarchy_config = hierarchy_config
+
+    def statistics_task_flow(self, abort_signal: Signal, fl_ctx: FLContext, statistic_task: str):
+        """Statistics task flow for the given task.
+
+        Args:
+            abort_signal: Abort signal.
+            fl_ctx: The FLContext.
+            statistic_task: Statistics task.
+        """
+        if self.hierarchy_config:
+            engine = fl_ctx.get_engine()
+            ws = engine.get_workspace()
+            app_conf_dir = ws.get_app_config_dir(fl_ctx.get_job_id())
+            hierarchy_config_file_path = os.path.join(app_conf_dir, self.hierarchy_config)
+            try:
+                with open(hierarchy_config_file_path) as hierarchy_config_file:
+                    hierarchy_config_json = json.load(hierarchy_config_file)
+            except FileNotFoundError:
+                self.system_panic(f"The hierarchy config file {hierarchy_config_file_path} does not exist.", fl_ctx)
+                return False
+            except IOError as e:
+                self.system_panic(
+                    f"An I/O error occurred while loading hierarchy config file {hierarchy_config_file_path}: {e}",
+                    fl_ctx,
+                )
+                return False
+            except json.decoder.JSONDecodeError as e:
+                self.system_panic(
+                    f"Failed to decode hierarchy config JSON from the file {hierarchy_config_file_path}: {e}", fl_ctx
+                )
+                return False
+            except Exception as e:
+                self.system_panic(
+                    f"An unexpected error occurred while loading hierarchy config file {hierarchy_config_file_path}: {e}",
+                    fl_ctx,
+                )
+                return False
+
+        self.log_info(fl_ctx, f"start prepare inputs for task {statistic_task}")
+        inputs = self._prepare_inputs(statistic_task)
+        results_cb_fn = self._get_result_cb(statistic_task)
+
+        self.log_info(fl_ctx, f"task: {self.task_name} statistics_flow for {statistic_task} started.")
+
+        if abort_signal.triggered:
+            return False
+
+        task_props = {StC.STATISTICS_TASK_KEY: statistic_task}
+        task = Task(name=self.task_name, data=inputs, result_received_cb=results_cb_fn, props=task_props)
+
+        self.broadcast_and_wait(
+            task=task,
+            targets=None,
+            min_responses=self.min_clients,
+            fl_ctx=fl_ctx,
+            wait_time_after_min_received=self.wait_time_after_min_received,
+            abort_signal=abort_signal,
+        )
+
+        self.global_statistics = get_global_stats(
+            self.global_statistics, self.client_statistics, statistic_task, hierarchy_config_json
```

Comment:
If self.hierarchy_config is empty/None, then hierarchy_config_json would be undefined.
It should be initialized to a default value before "if self.hierarchy_config:" or placed in the "else" clause.

Reply:
Done, I have added else that reports error if `self.hierarchy_config` is empty/None and returns. Hoping it looks ok now.

Reply:
Looks good.